thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/SubscriptionGroupManagerImpl.java

```diff
@@ -16,15 +16,36 @@
 import ai.startree.thirdeye.datalayer.dao.GenericPojoDao;
 import ai.startree.thirdeye.spi.datalayer.bao.SubscriptionGroupManager;
 import ai.startree.thirdeye.spi.datalayer.dto.SubscriptionGroupDTO;
+import com.codahale.metrics.CachedGauge;
+import com.codahale.metrics.MetricRegistry;
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
+import java.util.concurrent.TimeUnit;
 
 @Singleton
 public class SubscriptionGroupManagerImpl extends
     AbstractManagerImpl<SubscriptionGroupDTO> implements SubscriptionGroupManager {
 
   @Inject
-  public SubscriptionGroupManagerImpl(GenericPojoDao genericPojoDao) {
+  public SubscriptionGroupManagerImpl(final GenericPojoDao genericPojoDao,
+      final MetricRegistry metricRegistry) {
     super(SubscriptionGroupDTO.class, genericPojoDao);
+    metricRegistry.register("subscriptionsCountTotal",
+        new CachedGauge<Integer>(15, TimeUnit.MINUTES) {
+      @Override
+      public Integer loadValue() {
+        return findAll().stream()
```

Comment:
if the method definition gets too long after the changes below. pull this out of constructor.

Reply:
Noted. I have created separate method named `registerMetrics()` at few places where I felt the constructor is looking too large. If it makes sense we can have this method everywhere by default (wherever metrics are defined in constructor) so that constructor looks clean?